crashtracker/src/crash_handler.rs

```diff
@@ -157,11 +197,34 @@ fn handle_posix_signal_impl(signum: i32) -> anyhow::Result<()> {
         "UNKNOWN"
     };
 
-    let pipe = receiver.stdin.as_mut().unwrap();
-    writeln!(pipe, "{DD_CRASHTRACK_BEGIN_SIGINFO}")?;
-    writeln!(pipe, "{{\"signum\": {signum}, \"signame\": \"{signame}\"}}")?;
-    writeln!(pipe, "{DD_CRASHTRACK_END_SIGINFO}")?;
+    writeln!(w, "{DD_CRASHTRACK_BEGIN_SIGINFO}")?;
+    writeln!(w, "{{\"signum\": {signum}, \"signame\": \"{signame}\"}}")?;
+    writeln!(w, "{DD_CRASHTRACK_END_SIGINFO}")?;
+    Ok(())
+}
 
+fn handle_posix_signal_impl(signum: i32) -> anyhow::Result<()> {
+    // Leak receiver, config, and metadata to avoid calling 'drop' during a crash
+    let receiver = RECEIVER.swap(ptr::null_mut(), SeqCst);
+    anyhow::ensure!(!receiver.is_null(), "No crashtracking receiver");
+    let receiver = unsafe { receiver.as_mut().context("")? };
+
+    let config = CONFIG.swap(ptr::null_mut(), SeqCst);
+    anyhow::ensure!(!config.is_null(), "No crashtracking config");
+    let (config, config_str) = unsafe { config.as_mut().context("")? };
+
+    let metadata_ptr = METADATA.swap(ptr::null_mut(), SeqCst);
+    anyhow::ensure!(!metadata_ptr.is_null());
```

Comment:
Minor: For consistency with the above approaches:

```suggestion
    anyhow::ensure!(!metadata_ptr.is_null(), "No metadata config");
```

Reply:
done